### ISSUE TITLE
[BUG] webkit page.evaluates sometimes hang

### ISSUE THREAD

**Issue by sammyyu** [microsoft/playwright#8570]
**Context:**
- Playwright Version: 1.14.0
- Operating System: linux v1.14.0-focal
- Node.js version: 14.17.5
- Browser: Webkit
- Extra: 
- 

**Code Snippet**
Going to https://www.buzzfeed.com/ the page evaluate call very frequently gets stuck
```
documentPageLength = await page.evaluate(
           () => document.documentElement.outerHTML.length
         );
```

**Describe the bug**
This exact same code works fine with firefox.  I would say 4/5 times it gets stuck and not on every web page.
I have attached the DEBUG=pw:api logs.  Please let me know if you need anything else.

**Comment by sammyyu** [comment]
[debug.log](https://github.com/microsoft/playwright/files/7080287/debug.log)

**Comment by pavelfeldman** [comment]
Works fine for me is there a small script I could run to reproduce?

**Comment by yury-s** [comment]
I can reproduce it. Sometimes it hangs inside evaluate in WebKit(both headless and headful), works fine in Chromium. In firefox the navigation times out.

**Comment by yury-s** [comment]
From the protocol log 'Runtime.callFunctionOn' does succeed but `page.evaluate()` will not return until we start closing the context.

**Comment by yury-s** [comment]
This is what's happening: `page.evaluate()` will [auto wait](https://playwright.dev/docs/navigations#auto-wait-1) for navigation if one is started during the eval. page.evaluate is implemented via several protocol commands, those which are relevant here are:
- Runtime.evaluate to evaluate utility script in the page
...
- Runtime.callFunctionOn that does actual evaluation of the user expression
Now, between the two evals we receive `Page.frameScheduledNavigation` which the page scheduled on its own but we attribute that to the page.evaluate and then wait for it to finish.

**Comment by pavelfeldman** [comment]
Let's collect more reports on this one to see if the fix is good.

**Comment by sammyyu** [comment]
Thanks for the fix, I can confirm the it is working in 1.15.0
